src/test/scala/org/broadinstitute/dsde/workbench/avram/api/FakeHttpServletRequest.scala

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.dsde.workbench.avram.api
+
+import java.io.BufferedReader
+import java.security.Principal
+import java.util
+import java.util.Locale
+
+import javax.servlet.http._
+import javax.servlet._
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+/**
+  * Stub implementations of header functions for HttpServletRequest.
+  */
+trait StubbedHeaders {
+  def headers: Map[String, List[String]]
+
+  def getHeader(name: String): String = {
+    Try(headers.filterKeys(_.equalsIgnoreCase(name))
```

Comment:
Potential alternative:
```
headers.filterKeys(_.equalsIgnoreCase(name)).values.flatten.headOption.getOrElse(null)
```

Reply:
Huh... I don't even remember writing that. I wonder if I copied it from stackoverflow or something. Anyway, IntelliJ suggested a change to yours of using `orNull` instead of `getOrElse(null)`.